### DISCUSSION TITLE
how to update user's email address with sending email verification code to the new email address?

### DISCUSSION THREAD

**Discussion by Metaildust** [serverpod/serverpod#1992] [Q&A] [answered]
At present, I understand that to update the user's email address, all you need to do is modify the userinfo database. But I would like to send a verification email to the old email address before that to confirm that the user wants to unbind the old email address, and then send a new email address to ensure that the new email address is available. But how should this custom verification email be created?

**Comment by SandPod** [comment, marked as answer]
Hey!

Currently, we don't have built-in support for this workflow in our authentication module.
But this is possible to implement yourself through your own endpoints.

A good starting point is to look at how the `createAccountRequest` [endpoint](https://github.com/serverpod/serverpod/blob/73f31ee132c71be3ce35720213abb04111c84e76/modules/serverpod_auth/serverpod_auth_server/lib/src/endpoints/email_endpoint.dart#L147) in combination with the `createAccount` [endpoint](https://github.com/serverpod/serverpod/blob/73f31ee132c71be3ce35720213abb04111c84e76/modules/serverpod_auth/serverpod_auth_server/lib/src/endpoints/email_endpoint.dart#L162) works in the email authentication provider.

There we go through the flow of validating an authentication code send to the user.

But this sounds like a good workflow to support in the module so feel free to submit a feature request for this workflow.
